### PR TITLE
[5.0] Events backward compatibility handling

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes
I have extracted b/c fixes from #41226 in to separated PR.

### Testing Instructions
Apply patch.
Edit `plugins/system/sef/src/Extension/Sef.php`, add `return true` at top of onAfterDispatch() function.

### Actual result BEFORE applying this Pull Request
An error, cannot set result for Imutable event

### Expected result AFTER applying this Pull Request
No error

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: IDK
- [x] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: IDK
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
Would it make sense to check if the $name is a string and numeric?

Actually only strings are allowed but we don't have a typehint for it...

**Comment by Fedik** [comment]
I think it is enought to check for "numeric", additional test for a string does not give us any extra value.

File: libraries/src/Event/AbstractEvent.php [hunk 1]
```diff
@@ -137,6 +137,23 @@ public function __construct(string $name, array $arguments = [])
...
+        // B/C check for numeric access to named argument, eg $event->getArgument('0').
+        if (is_numeric($name)) {
+            if (key($this->arguments) != 0) {
+                $argNames = \array_keys($this->arguments);
+                $name     = $argNames[$name] ?? '';
+            }
+
+            @trigger_error(
+                sprintf(
+                    'Numeric access to named event arguments is deprecated, and will not work in Joomla 6. Event %s argument %s',
```
Comment (laoneo) [inline libraries/src/Event/AbstractEvent.php:149]: According to our policy, the removal should be for 7.0.

**Comment by HLeithner** [comment]
I would do this in 4.4 because it's not an intended behaviour (if we remove it in 6 then is another story)

**Comment by Fedik** [comment]
I can redo it to 4.4, no problem

File: libraries/src/Event/AbstractEvent.php (hunk 1 above)
Comment (Fedik) [reply libraries/src/Event/AbstractEvent.php:149]: It not a feature, in missues of the event api.
It should not exist, at all.

**Comment by Fedik** [comment]
PR for 4.4 #41357

**Pull request closed without merging**